### ISSUE TITLE
Cannot Install Oh-My-Zsh on centos 6 x64

### ISSUE THREAD

**Issue by ghost** [ohmyzsh/ohmyzsh#8533]
**Describe the bug**

I cannot install oh my zsh in either way via wget or curl. I've git and zsh installed. Here are the errors I got.
```bash
[root@openvpn ~]# curl -Lo install.sh https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8074  100  8074    0     0   121k      0 --:--:-- --:--:-- --:--:-- 7884k
[root@openvpn ~]# sh install.sh
Cloning Oh My Zsh...
error: unknown switch `c'
usage: git clone [options] [--] <repo> [<dir>]

    -v, --verbose         be more verbose
    -q, --quiet           be more quiet
    --progress            force progress reporting
    -n, --no-checkout     don't create a checkout
    --bare                create a bare repository
    --mirror              create a mirror repository (implies bare)
    -l, --local           to clone from a local repository
    --no-hardlinks        don't use local hardlinks, always copy
    -s, --shared          setup as shared repository
    --recursive           initialize submodules in the clone
    --template <path>     path the template repository
    --reference <repo>    reference repository
    -o, --origin <branch>
                          use <branch> instead of 'origin' to track upstream
    -b, --branch <branch>
                          checkout <branch> instead of the remote's HEAD
    -u, --upload-pack <path>
                          path to git-upload-pack on the remote
    --depth <depth>       create a shallow clone of that depth

Error: git clone of oh-my-zsh repo failed
[root@openvpn ~]# ^C
```

**System:**
 - OS: CentOS 6 x64
 - Zsh version: zsh 4.3.11 (x86_64-redhat-linux-gnu)
 - Terminal emulator: xterm-256color

**Comment by mcornella** [comment]
What git version do you have installed?

**Comment by paradiseduo** [comment]
i have the same issue with:
```
[root@host upload]# git --version
git version 1.7.1
```

**Comment by paradiseduo** [comment]
```
[root@host upload]# bash install.sh
Cloning Oh My Zsh...
error: unknown switch `c'
usage: git clone [options] [--] <repo> [<dir>]

    -v, --verbose         be more verbose
    -q, --quiet           be more quiet
    --progress            force progress reporting
    -n, --no-checkout     don't create a checkout
    --bare                create a bare repository
    --mirror              create a mirror repository (implies bare)
    -l, --local           to clone from a local repository
    --no-hardlinks        don't use local hardlinks, always copy
    -s, --shared          setup as shared repository
    --recursive           initialize submodules in the clone
    --template <path>     path the template repository
    --reference <repo>    reference repository
    -o, --origin <branch>
                          use <branch> instead of 'origin' to track upstream
    -b, --branch <branch>
                          checkout <branch> instead of the remote's HEAD
    -u, --upload-pack <path>
                          path to git-upload-pack on the remote
    --depth <depth>       create a shallow clone of that depth

Error: git clone of oh-my-zsh repo failed
```

**Comment by mcornella** [comment]
`-c` in git clone was introduced in 1.7.7, 8 years ago. I'll see if I can code it some other way that still doesn't cause a regression for the fsckobjects errors. In the meantime, is there a possibility of using a newer git version?

**Comment by paradiseduo** [comment]
> `-c` in git clone was introduced in 1.7.7, 8 years ago. I'll see if I can code it some other way that still doesn't cause a regression for the fsckobjects errors. In the meantime, is there a possibility of using a newer git version?

After update git, this issue has gone

**Comment by TriMoon** [comment]
@20191229somesysadmin , @paradiseduo 
If the default Cent OS 6 has git version <=1.7.7 in its distribution i would suggest to contact them about it :wink: 
The latest version via Kubuntu 19.10 (my system) has `git version 2.20.1`...

**Comment by larson-carter** [comment]
I figured out a work around! It is an issue with the version of git that you have installed!

Here is a guide on how to install ohmyzsh from a fresh centos install!

1. `sudo yum install zsh`
2. `sudo yum install wget git`
3. `sudo yum remove git*`
4. This next step depends on your CentOS version - pay atention to the URL
* CentOS 6.x `sudo yum install http://opensource.wandisco.com/centos/6/git/x86_64/wandisco-git-release-6-1.noarch.rpm`
* CentOS 7.x `sudo yum install http://opensource.wandisco.com/centos/7/git/x86_64/wandisco-git-release-7-1.noarch.rpm`
* CentOS 8.x `sudo yum install http://opensource.wandisco.com/centos/7/git/x86_64/wandisco-git-release-7-2.noarch.rpm`
5. `sudo yum install git`
6. `git --version`
* This should show `git version 2.22.0` or higher
7. `sudo curl -Lo install.sh https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh`
8. `sh install.sh`
* Then follow the prompts to change your default shell, etc
9. Configure ohmyzsh

**Comment by mcornella** [comment]
Closing this since Centos 6 is dead.

**Comment by ThemysciraData** [comment]
While CentOS / RHEL 6 is "dead" unfortunately, some of us are stuck with it and unable to upgrade git because of no root access. Been trying several work arounds, and just going to keep plugging away I guess.

**Comment by mcornella** [comment]
You can install Oh My Zsh manually by following [the steps here](https://github.com/ohmyzsh/ohmyzsh#manual-installation) but using the zip of the repo instead of `git clone`ing:
```
# from home directory
wget https://github.com/ohmyzsh/ohmyzsh/archive/refs/heads/master.zip
unzip master.zip
mv ohmyzsh-master .oh-my-zsh
```
and following the rest of the steps normally.

We could support this use case as a new feature, because I don't think there's a way to rewrite the git clone call to support git < 1.7.2.

**Comment by ThemysciraData** [comment]
Ok. I was mostly there then. I just was trying to edit the install.sh in /tools to work. Not sure why I didn't just copy it over like you said, probably distractions from meetings etc.

**Comment by mcornella** [comment]
I've pushed #10621 to allow using older git versions, can you please try the installer from `https://raw.githubusercontent.com/mcornella/ohmyzsh/fix/installer-support-git-v1.7.1/tools/install.sh`?
